pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java

```diff
@@ -266,7 +266,7 @@ protected void initializeClientConfiguration() {
             this.casPrefixUrl += "/";
         }
         if (CommonHelper.isBlank(this.casPrefixUrl)) {
-            this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login", "/");
+            this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login$", "/$");
```

Comment:
We'll need to be smart enough to only replace the "/login" path element and not the one in the actual domain. Asking folks to define their CAS login url with a $ at the end is not the way to go. Parse the path correctly, and take out the bit that is actually needed to be removed rather than relying on the deployer to correctly define it for you.